app.js

```diff
@@ -4,186 +4,186 @@
+            const prefix = metrics[0].slice(1);
+            if (prefix !== "CVSS:4.0") {
+                console.log("Error invalid vector, missing CVSS v4.0 prefix");
+                return;
+            }
+            metrics.shift();
 
-                expected = Object.entries(this.expectedMetricOrder)[oi++]
+            const toSelect = {};
+            let oi = 0;
+            for (const index in metrics) {
+                const [key, value] = metrics[index].split(":");
+                let expected = Object.entries(this.expectedMetricOrder)[oi++];
                 while (true) {
                     // If out of possible metrics ordering, it not a valid value thus
                     // the vector is invalid
-                    if (expected == undefined) {
-                        console.log("Error invalid vector, too many metric values")
-                        return
+                    if (expected === undefined) {
+                        console.log("Error invalid vector, too many metric values");
+                        return;
                     }
-                    if (key != expected[0]) {
+                    if (key !== expected[0]) {
                         // If not this metric but is mandatory, the vector is invalid
                         // As the only mandatory ones are from the Base group, 11 is the
                         // number of metrics part of it.
                         if (oi <= 11) {
-                            console.log("Error invalid vector, missing mandatory metrics")
-                            return
+                            console.log("Error invalid vector, missing mandatory metrics");
+                            return;
                         }
                         // If a non-mandatory, retry
-                        expected = Object.entries(this.expectedMetricOrder)[oi++]
-                        continue
+                        expected = Object.entries(this.expectedMetricOrder)[oi++];
+                        continue;
                     }
-                    break
+                    break;
                 }
                 // The value MUST be part of the metric's values, case insensitive
                 if (!expected[1].includes(value)) {
-                    console.log("Error invalid vector, for key " + key + ", value " + value + " is not in " + expected[1])
-                    return
+                    console.log(`Error invalid vector, for key ${key}, value ${value} is not in ${expected[1]}`);
+                    return;
                 }
-                if (key in this.cvssSelected) {
-                    toSelect[key] = value
+                if (key in this.vectorMetrics) {
+                    toSelect[key] = value;
                 }
             }
-
-            // Apply iff is compliant
-            for (key in toSelect) {
-                this.cvssSelected[key] = toSelect[key]
+            
+            // Apply if it is compliant
```

Comment:
iif = if and only if

Reply:
That was unclear, i am going to write it clearly